src/util/StringUtils.h

```diff
@@ -588,22 +588,20 @@ string decodeUrl(const string& url) {
       } else if (h1 >= 'a' && h1 <= 'f') {
         h1 = h1 - 'a' + 10;
       } else {
-        h1 = -1;
+        decoded += '%';
```

Comment:
Is this indeed valid URL encoding? As I see it you interpret % as % when the next character cannot be interpreted as HEX. Is there a reason for this?

Reply:
I see, you just refactored the original behaviour. I'm still not sure whether this is within the specification.

Reply:
Neither am I. It's a bit hackish too.